developer_manual/app/tutorial/routes_and_controllers.rst

```diff
@@ -0,0 +1,210 @@
+            parent::__construct($AppName, $request);
+        }
+
+        /**
+         * @NoAdminRequired
+         * @NoCSRFRequired
+         */
+        public function index() {
+            // Renders ownnotes/templates/main.php
+            return new TemplateResponse('ownnotes', 'main');
+        }
+
+    }
+
+The first argument to the constructor specifies which application’s template directory to search.
+The second argument specifies the template to use, minus file extension (``.php``).
+Templates are, effectively, not much more than the original PHP files, which were a combination of PHP and HTML.
+   
+.. note::
+   The ``OCP`` namespace maps to ``ownCloud/core/lib/public``.
+
+.. note:: 
+   The ``@NoAdminRequired`` and ``@NoCSRFRequired`` annotations in index's docblock above turn off security checks, as they're not necessary for this method. 
+   See :doc:`../fundamentals/controllers` for more information.
+
+With an initial overview of controllers (and templates) completed, we'll now create the core of a controller which handles AJAX requests for the application. 
+Create a new controller, called ``ownnotes/lib/Controller/NoteController.php``, with the following content:
+
+.. code-block:: php
+
+   <?php
+    namespace OCA\OwnNotes\Controller;
+
+    use OCP\IRequest;
+    use OCP\AppFramework\Controller;
+
+    class NoteController extends Controller {
+
+        public function __construct($AppName, IRequest $request){
+            parent::__construct($AppName, $request);
+        }
+
+        /**
+         * @NoAdminRequired
+         */
+        public function index() {
+            // empty for now
+        }
+
+        /**
+         * @NoAdminRequired
+         *
+         * @param int $id
+         */
+        public function show($id) {
+            // empty for now
+        }
+
+        /**
+         * @NoAdminRequired
+         *
+         * @param string $title
+         * @param string $content
+         */
+        public function create($title, $content) {
+            // empty for now
+        }
+
+        /**
+         * @NoAdminRequired
+         *
+         * @param int $id
+         * @param string $title
+         * @param string $content
+         */
+        public function update($id, $title, $content) {
+            // empty for now
+        }
+
+        /**
+         * @NoAdminRequired
+         *
+         * @param int $id
+         */
+        public function destroy($id) {
+            // empty for now
+        }
+
+    }
+
+You can see that it's largely the same as the ``PageController``, but with a range of CRUD methods.
+Take special note of ``show``, ``create``, ``update``, and ``destroy``.
+The parameters to these functions are extracted from the request body and the URL, using the controller method's variable names. 
+
+.. note::
+   Since PHP does not support type hints for primitive types, such as ints and booleans, we need to add them as annotations in the comments. 
```

Comment:
From php 7 onwards there are scalar type hints - in the preface for the tutorial we should mention that the tutorial is gathered towards development in 7+

Reply:
Fair point.